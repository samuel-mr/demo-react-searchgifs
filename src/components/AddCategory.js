import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategory }) => {
    const [value, setValue] = useState('')

    const onInputChange = (e) => {
        setValue(e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        if (value.trim().length > 2) {
            setCategory(currentValue => [value, ...currentValue])
            setValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{value}</p>
            <input type='text' value={value} onChange={onInputChange}></input>

        </ form >
    )

}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}