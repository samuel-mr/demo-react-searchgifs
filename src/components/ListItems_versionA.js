import React, { useState, useEffect } from 'react'
import { ListItemImage } from './ListItemImage'
import { getGifs } from '../helpers/getGifs'

export const Listitem = ({ category }) => {

    const [images, setImages] = useState([])

    useEffect(() => {
        getGifs(category)
            .then(imgs => setImages(imgs))
    }, [category])


    return (
        <>
            <h1>{category}</h1>
            <div className='card-grid'>
                {
                    images.map(item => (
                        <ListItemImage
                            key={item.id}
                            {...item}
                        />
                    ))
                }
            </div>
        </>
    )
}