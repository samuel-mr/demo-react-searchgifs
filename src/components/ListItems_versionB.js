import React from 'react'
import { ListItemImage } from './ListItemImage'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const ListItems_versionB = ({ category }) => {

    const { data:images, loading } = useFetchGifs(category)


    return (
        <>
            <h1 className='animate__animated animate__slideInDown'>{category}</h1>
            {loading && <div className='animate__animated animate__flash'>Cargando... </div>}
            <div className='card-grid'>
                {
                    images.map(item => (
                        <ListItemImage
                            key={item.id}
                            {...item}
                        />
                    ))
                }
            </div>
        </>
    )
}