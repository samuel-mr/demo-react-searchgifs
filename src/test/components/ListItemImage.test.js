import React from 'react'
import { shallow } from 'enzyme'
import { ListItemImage } from "../../components/ListItemImage"

describe('Testing Components', () => {
    const wrapper = shallow(<ListItemImage title='hola' url='http://direccion' />)

    test('debe igualar el snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe mostrar un titulo si le agrego uno', () => {
        const parrafo = wrapper.find('p').text()
        expect(parrafo).toBe('hola')
    })
    test('debe tener una imagen con url y alt', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe('http://direccion')
        expect(img.prop('alt')).toBe('hola')
    })

    test('debe mostrar la clase de animacion dentro del contendor div', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')

        expect(className.includes('animate__animated')).toBeTruthy()
    })
})