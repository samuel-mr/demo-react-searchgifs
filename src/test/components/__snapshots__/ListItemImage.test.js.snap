// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components debe igualar el snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="hola"
    src="http://direccion"
  />
  <p>
    hola
  </p>
</div>
`;
